src/swarm.js

```diff
@@ -47,6 +47,12 @@ function Swarm (peerInfo) {
     // set up the transport and add the list of incoming streams
     // add transport to the list of transports
 
+    var multiaddr = options.multiaddr
+    if (multiaddr) {
+      // no need to pass that to the transports
+      delete options.multiaddr
+    }
+
```

Comment:
ok :) (although in the future, things like libp2p-webrtc might like that multiaddr, let's see, I might bring it back)